airflow-core/src/airflow/utils/hitl_shared_links.py

```diff
@@ -0,0 +1,220 @@
+        if not self.secret_key:
+            raise ValueError("HITL shared link secret key is not configured")
+
+        signature = hmac.new(
+            self.secret_key.encode("utf-8"), payload.encode("utf-8"), hashlib.sha256
+        ).digest()
+        return base64.urlsafe_b64encode(signature).decode("utf-8")
+
+    def _verify_signature(self, payload: str, signature: str) -> bool:
+        """Verify HMAC signature for the payload."""
+        expected_signature = self._generate_signature(payload)
+        return hmac.compare_digest(expected_signature, signature)
+
+    def generate_link(
+        self,
+        dag_id: str,
+        dag_run_id: str,
+        task_id: str,
+        map_index: int | None = None,
+        link_type: str = "action",
+        action: str | None = None,
+        expires_in_hours: int | None = None,
+        base_url: str | None = None,
+    ) -> dict[str, Any]:
+        """
+        Generate a shared link for HITL task.
+
+        :param dag_id: DAG ID
+        :param dag_run_id: DAG run ID
+        :param task_id: Task ID
+        :param map_index: Map index for mapped tasks
+        :param link_type: Type of link ('action' or 'redirect')
+        :param action: Action to perform (for action links)
+        :param expires_in_hours: Custom expiration time in hours
+        :param base_url: Base URL for the link
+        """
+        if not self.is_enabled():
+            raise ValueError("HITL shared links are not enabled")
+
+        if link_type == "action" and not action:
+            raise ValueError("Action is required for action-type links")
+
+        expiration_hours = expires_in_hours or self.default_expiration_hours
+        expires_at = timezone.utcnow() + timedelta(hours=expiration_hours)
+
+        payload_data = {
+            "dag_id": dag_id,
+            "dag_run_id": dag_run_id,
+            "task_id": task_id,
+            "map_index": map_index,
+            "link_type": link_type,
+            "action": action,
+            "expires_at": expires_at.isoformat(),
+        }
+
+        payload_str = json.dumps(payload_data, sort_keys=True)
+        signature = self._generate_signature(payload_str)
+
+        encoded_payload = base64.urlsafe_b64encode(payload_str.encode("utf-8")).decode("utf-8")
+
+        if base_url is None:
+            base_url = conf.get("api", "base_url", fallback="http://localhost:8080")
+
+        if map_index is not None:
+            url_path = f"/api/v2/hitl-details/api/v2/hitl-details-share-link/{dag_id}/{dag_run_id}/{task_id}/{map_index}"
+        else:
+            url_path = f"/api/v2/hitl-details/api/v2/hitl-details-share-link/{dag_id}/{dag_run_id}/{task_id}"
+
+        query_params = {
+            "payload": encoded_payload,
+            "signature": signature,
+        }
+
+        link_url = f"{base_url.rstrip('/')}{url_path}?{urlencode(query_params)}"
+
+        return {
+            "task_instance_id": f"{dag_id}.{dag_run_id}.{task_id}.{map_index or -1}",
```

Comment:
Let's do the actual id, or provide each component of the logical key separately.